Add more info on correlating trace and container metrics (#28870)
DataDog/documentation: content/en/opentelemetry/integrations/docker_metrics.md

{{< /tabs >}}

## Correlate traces with container metrics

> To correlate trace and container metrics, configure [Universal Service Monitoring attributes][4] for each service, and set the following resource attributes for each service: 

Reviewer: USM does not provide any trace information

Author: Oh, I didn't even consider that. This text was already on the page, so it must be incorrect. I'll remove mention of USM from here. Thanks!

To populate these attributes:
- **Using the Collector**: The `docker_stats` receiver automatically sets these attributes for you.
- **Using the Datadog Agent**: Add a container resource detector in your application code.  
   For example, using Go:
   ```go
   // resource.WithContainer() adds container.id so the Agent can tag the spans
   res, err := resource.New(
       ctx,
       resource.WithContainer(),                    
       resource.WithFromEnv(),
       semconv.ServiceNameKey.String("calendar"),   
   )
   ```

See the complete example in [opentelemetry-examples][8].

## Data collected

The Docker Stats receiver generates container metrics for the OpenTelemetry Collector. The Datadog Exporter translates container metrics to their Datadog counterparts for use in the following views:

- [Containers Overview default dashboard][6]
- [APM Trace view][7] with container metrics

Learn more about [mapping between OpenTelemetry and Datadog semantic conventions for resource attributes][5].

> The following table shows what Datadog container metric names are associated with corresponding OpenTelemetry container metric names:

Reviewer 2: ```suggestion
The following table shows the Datadog container metric names that correspond to OpenTelemetry container metric names:
```
Does this keep the meaning?

Author: Yeah, this is better. Thanks!

[4]: /universal_service_monitoring/setup/
[5]: /opentelemetry/guide/semantic_mapping/
[6]: /opentelemetry/otel_collector_datadog_exporter/?tab=onahost#containers-overview-dashboard
[7]: /tracing/trace_explorer/trace_view/

> [8]: https://github.com/DataDog/opentelemetry-examples/blob/main/apps/rest-services/golang/calendar/main.go#L133

Reviewer 3: This line changed since we last updated this example, I'd recommend to use a pinned version of the repo when linking lines of code.

Author: Good catch. Thanks for taking a look!
